### PR TITLE
[Wave Matrix] Improve stability of wave matrix tests

### PR DESCRIPTION
**Pull request by bob80905**
The wave matrix tests had a remaining __debug_break() call left in the code.
There was also a function that risked returning the wrong comparison value due to underflow behavior.
This PR improves the stability of the wave matrix tests by removing the __debug_break() call and preventing underflow from happening.
Fixes https://github.com/microsoft/DirectXShaderCompiler/issues/5498
Fixes https://github.com/microsoft/DirectXShaderCompiler/issues/5369

### CODE REVIEW

File: tools/clang/unittests/HLSLExec/ExecutionTest.cpp [hunk 1]
```diff
@@ -6433,20 +6433,29 @@ HRESULT TableParameterHandler::ParseTableRow() {
   return S_OK;
 }
 
-static bool CompareOutputWithExpectedValueInt(int output, int ref, int tolerance) {
-    return output - ref <= tolerance && ref - output <= tolerance;
+static bool CompareOutputWithExpectedValueInt(int output, int ref,
+                                              int tolerance) {
+  return output - ref <= tolerance && ref - output <= tolerance;
```
Comment (python3kgae) [inline tools/clang/unittests/HLSLExec/ExecutionTest.cpp:6438]: Could we change to ((output - ref) <= tolerance) && ((ref - output) <= tolerance); ?
It is hard to follow now.

**Review by python3kgae — APPROVED** [review]

**Review by pow2clk — APPROVED** [review]
The changes are fine. We are trying to have a 1:1 mapping of PRs to bugs where possible. I realize that these were both pretty small in terms of lines of code, but it would also have been easy to separate them that way. I wouldn't suggest doing it now though. Just a note for the future
